File: src/main/java/com/ljl/ai/model/dto/ChatResponse.java
package com.ljl.ai.model.dto;

import com.ljl.ai.model.entity.KnowledgeSource;
import com.ljl.ai.model.entity.ToolInvocation;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 对话响应DTO
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatResponse {
    
    /**
     * 会话ID
     */
    private String sessionId;
    
    /**
     * 消息ID
     */
    private String messageId;
    
    /**
     * AI回复内容
     */
    private String content;
    
    /**
     * 响应时间
     */
    private LocalDateTime responseTime;
    
    /**
     * 知识来源列表（用于溯源）
     */
    private List<KnowledgeSource> knowledgeSources;
    
    /**
     * 工具调用记录
     */
    private List<ToolInvocation> toolInvocations;
    
    /**
     * 诊断结果（如有）
     */
    
    /**
     * 是否成功
     */
    private Boolean success;
    
    /**
     * 错误信息
     */
    private String errorMessage;
}

File: frontend/src/chatResponse.js
// ChatResponse.content 是面向用户的降级说明；通用错误响应则回退到 errorMessage/error。
export function chatFailureMessage(data) {
  const candidates = [
    ...(data?.success === false ? [data.content] : []),
    data?.errorMessage,
    data?.error,
    data?.message
  ]
  return (
    candidates.find((value) => typeof value === 'string' && value.trim())?.trim() || '接口请求失败'
  )
}
